components/eamxx/cime_config/eamxx_buildnml.py

```diff
@@ -249,6 +249,63 @@ def _dict_representer(dumper, data):
     else:
         return yaml.dump(data, item, OrderedDumper, **kwds)
 
+###############################################################################
+def evaluate_selector(sel_name, sel_value, ez_selectors, case, child_name):
+###############################################################################
+    # Parse and and ors into separate statements
+    and_syntax = " @@ "
+    or_syntax  = " || "
+    statements = []
+    if and_syntax in sel_value:
+        expect(or_syntax not in sel_value, "Cannot mix @@ and ||")
+        statements = sel_value.split(and_syntax)
+    elif or_syntax in sel_value:
+        statements = sel_value.split(or_syntax)
+    else:
+        statements = [sel_value]
+
+    # Get relevant value from case
+    if sel_name in ez_selectors:
+        ez_env, ez_regex = ez_selectors[sel_name]
+        case_val = case.get_value(ez_env)
+        expect(case_val is not None,
+              "Bad easy selector '{}' definition. Relies on unknown case value '{}'".format(sel_name, ez_env))
+
+        ez_regex_re = re.compile(ez_regex)
+        m = ez_regex_re.match(case_val)
+        if m:
+            groups = m.groups()
+            expect(len(groups) == 1,
+                    "Selector '{}' has invalid custom regex '{}' which does not capture exactly 1 group".format(sel_name, ez_regex))
+            val = groups[0]
+        else:
+            # If the regex doesn't even match the case val, make the value so it won't match anything
+            val = None
+
+    else:
+        val = case.get_value(sel_name)
+        expect(val is not None,
+               "Bad selector '{0}' for child '{1}'. '{0}' is not a valid case value or easy selector".format(sel_name, child_name))
+
+    # Check value for matches with statements
+    result = False if or_syntax in sel_value else True
+    for statement in statements:
+        is_negation = statement.startswith("!")
+        if is_negation:
+            statement = statement.lstrip("!")
+
+        val_re = re.compile(statement)
+
+        curr_result = not(val is None or (val_re.match(val) is None and not is_negation) or (val_re.match(val) is not None and is_negation))
```

Comment:
How about splitting this line into 2-3, for clarity? Like
```py
found_match = val_re.match(val) is not None # check if regex yielded a match
desired_match = not is_negation # whether we want to match regex or not
match_is_good= found_match==desired_match # check if the match was desired or not
curr_result = val is not None and match_is_good
```

Reply:
Good idea, done. I also had the entire function just return False if the easy selector fails to get a value.